### DISCUSSION TITLE
have you all thought about adding urlon support in curl?

### DISCUSSION THREAD

**Discussion by ghost** [curl/curl#9655] [Q&A] [answered]
Hi all.

I recently read that curl supports [json](https://github.com/curl/curl/wiki/JSON). However, there is a json based url pattern which from what I understand curl still doesn't support. So... I would like to know if it is possible... add urlon support in curl... *have you all thought about adding urlon support in curl?*

#### 1. What is urlon?
URL Object Notation - [URLON](https://github.com/Double-oxygeN/urlon-nim).

#### 2. sample
```ruby
curl --get --json --urlon http://website.com/#{"table":{"achievement":{"column":"instance","ascending":true}}}
```

#### 3. commands
- ``"curl --get --urlon --json"``: get value from urlon
- ``"curl --post --urlon --json"``: post value from urlon
- ``"curl --delete --urlon --json"``: delete value from urlon

**Comment by bagder** [comment, marked as answer]
This looks like someone's custom way to write structured (json) data in the fragment portion of a URL. Since the data is already there in the URL, it would be easy to use a separate tool to just convert that into plain json. I see no official spec for this format.

Personally, I don't feel a pressing need nor desire from users to have this supported.
